fhttp/http.go

```diff
@@ -395,21 +395,12 @@ func NewFastClient(o *HTTPOptions) Fetcher {
 		bc.port = url.Scheme // ie http which turns into 80 later
 		log.LogVf("No port specified, using %s", bc.port)
 	}
-	addrs, err := net.LookupIP(bc.hostname)
-	if err != nil {
-		log.Errf("Unable to lookup '%s' : %v", bc.host, err)
-		return nil
-	}
-	if len(addrs) > 1 && log.LogDebug() {
-		log.Debugf("Using only the first of the addresses for %s : %v", bc.host, addrs)
-	}
-	log.Debugf("Will go to %s", addrs[0])
-	bc.dest.IP = addrs[0]
-	bc.dest.Port, err = net.LookupPort("tcp", bc.port)
-	if err != nil {
-		log.Errf("Unable to resolve port '%s' : %v", bc.port, err)
+	addr := fnet.Resolve(bc.hostname, bc.port)
```

Comment:
being pedantic :)
go idiom is to return err and check for it
```
- addr := fnet.Resolve(bc.hostname, bc.port)
- if addr == nil {
+ addr, err := fnet.Resolve(bc.hostname, bc.port)
+ if err != nil {
```

Reply:
I know; but I think nil is also a perfectly valid marker for failure - with the assumption (which I realize is 'bad' for a library, I have a comment to that effect in fnet about splitting the functions / error handling - specially where I have log.Fatalf...) that error logging already happened